File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // n is height, h is hash, d is space, and i is row.
    // Get integer between 1 and 8 inclusively.
    int n;
    do
    {
        n = get_int("Please enter the height of the wall: ");
    }
    while (n > 8 || n < 1);

    // Create n rows.
    for (int i = 0; i < n; i++)
    {
        // Print  blank spaces " " (n - 1 - i) times
        for (int d = n - 1; d > i; d--)
        {
            printf(" ");
        }

        // Print right aligned hashes '#' (i + 1) times.
        for (int h = -1; h < i; h++)
        {
            printf("#");
        }

        printf(" ");
        // Print left aligned hashes '#' (i + 1) times.
        for (int h = -1; h < i; h++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: sentimental-mario-more/mario.py
from cs50 import get_int

# Prompt the user to enter the height until a valid input is recieved
height = 0
while height < 1 or height > 8:
    height = get_int("Height: ")

# Iterate over each row of the pyramid
for row in range(height):
    # Print spaces for the left side of the left-aligned pyramid
    print((height - 1 - row) * " ", end = "")

    # Print hashes for the left side of the left-aligned pyramid
    print((row + 1) * "#", end = "")

    # Print spaces between the two pyramids
    print("  ", end = "")

    # Print hashes for the right side of the right-aligned pyramid
    print((row + 1) * "#")
